src/Microsoft.Health.SqlServer/ManagedIdentitySqlConnection.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Data.SqlClient;
+using Microsoft.Health.SqlServer.Configs;
+
+namespace Microsoft.Health.SqlServer
+{
+    public class ManagedIdentitySqlConnection : ISqlConnection
+    {
+        private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
+        private readonly IAccessTokenHandler _accessTokenHandler;
+
+        public ManagedIdentitySqlConnection(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, IAccessTokenHandler accessTokenHandler)
+        {
+            EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
+            EnsureArg.IsNotNull(accessTokenHandler, nameof(accessTokenHandler));
+
+            _sqlServerDataStoreConfiguration = sqlServerDataStoreConfiguration;
+            _accessTokenHandler = accessTokenHandler;
+        }
+
+        /// <summary>
+        /// Get sql connection after getting access token.
+        /// </summary>
+        /// <param name="connectToMaster">Should connect to master?</param>
```

Comment:
I wonder if we should change the name of this to something like `clearInitialCatalog` to be more transparent as to what happens. Alternatively, we could change this to property to be `initialCatalog` and it could allow you to set it to whatever you wanted.

Reply:
I agree. I will change it to `clearInitialCatalog` such that it can still fetch the initial catalog from connection string as before.

Reply:
Updated to accept `initialCatalog`.